app/recurring_job.go

```diff
@@ -359,6 +393,16 @@ func (job *Job) run() (err error) {
 		job.logger.Infof("Volume %v is in state %v", volumeName, volume.State)
 	}
 
+	if len(volume.RebuildStatus) != 0 {
+		isForceCreateSnapshot, err := getIsForceCreateSnapshot(job)
+		if err != nil {
+			return err
+		}
+		if !isForceCreateSnapshot {
+			return fmt.Errorf("recurring job will not start when replica is rebuilding")
+		}
+	}
```

Comment:
Should we always reject it when the option is enabled, or should we consider the number of snapshots and the retaintion number?

Reply:
Removed this part, we will do deleting first or do the recurring job like before.
Thank you.